src/Symfony/Component/Lock/StoreInterface.php

```diff
@@ -52,6 +55,8 @@ public function putOffExpiration(Key $key, $ttl);
 
     /**
      * Removes a resource from the storage.
+     *
+     * @throws LockReleasingException
```

Comment:
right now `Lock::release()`  does not seem to expect this method to throw
https://github.com/symfony/symfony/blob/05ebca763e25e055b9826caa141519bf031dd37b/src/Symfony/Component/Lock/Lock.php#L148-L157
I would revert this and stop rethrowing in `ZookeeperStore` as any userland `LockInterface` implem probably behaves the same

Reply:
Thanks for the comment. I see this as a nice second line of defense in case the store doesn't throw exception. IMO the stores must throw the exception, because in the above case, the original exception message is masked. I don't see this to be a problem if the stores throw `LockReleasingException` instead going forward.

Reply:
I agree that rethrowing in the store might be useful. 
My reasoning is that `Lock` catches exceptions thrown by other methods of StoreInterface for updating its state + logging some clean messages before eventual rethrow, see e.g. https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Lock/Lock.php#L127-L129. 
Not doing the same in `release()` feels inconsistent to me, shouldn't we add some `try/catch` around the `delete()` call and behave the same? cc @jderusse @nicolas-grekas

Reply:
Yes, I did see the logging part, but it's a `noop` eventually and so was not sure how it benefits. I would definitely go for this over consistency in this particular case. Happy to hear other's thoughts as well.

Reply:
indeed, other methods:
 - log failures to help for debugging
 - update the `dirty` state )=> but this should not be the case for exception thrown by a call to delete
 - wrap generic exception.

For consistency we could wrap generic exception and log failure (I'm not really convinced by the value of those log :-/)

Anyway, IMHO this is not the purpose of this PR, I suggest to open a dedicated PR with something like
```php
    public function release()
    {
        try {
            $this->store->delete($this->key);
        } catch (LockReleasingException $e) {
            throw $e
        } catch (\Exception $e) {
            throw new LockReleasingException(sprintf('Failed to release the "%s" lock.', $this->key), 0, $e);
        }
        $this->dirty = false;
        if ($this->store->exists($this->key)) {
            $this->logger->notice('Failed to release the "{resource}" lock.', array('resource' => $this->key));
            throw new LockReleasingException(sprintf('Failed to release the "%s" lock.', $this->key));
        }
    }
```